File: modules/client-java/src/main/java/be/vlaanderen/awv/atom/java/EntryConsumer.java
package be.vlaanderen.awv.atom.java;

import be.vlaanderen.awv.atom.format.Entry;
import be.vlaanderen.awv.atom.FeedPosition;
import be.vlaanderen.awv.atom.FeedProcessingException;

public interface EntryConsumer<E> {
    void accept(FeedPosition position, Entry<E> entry) throws FeedProcessingException;
}

File: modules/client-scala/src/main/scala/be/wegenenverkeer/atom/EntryConsumer.scala
package be.wegenenverkeer.atom

import be.wegenenverkeer.atomium.format.Entry

import scala.util.Try

/**
 * An entry consumer is responsible for consuming the new entries offered by the feed processor.
 *
 * The entry consumer should handle the new event and persist the current feed position (possibly in 1 database
 * transaction).
 *
 * @tparam E the type of the entries in the feed
 */
@deprecated
trait EntryConsumer[E] extends ((Entry[E]) => Try[Entry[E]])
